src/components/Map/mapUtil.ts

```diff
@@ -12,6 +13,68 @@ export type ClusterClickHandlerType = (
   facilites: GeojsonFeatureType<MinimalRecordType>[]
 ) => void
 
+interface GetFilteredFacilitiesPropType {
+  facilities: MinimalRecordType[]
+  labels: GristLabelType[]
+}
+
+interface IsFaclilityActivePropType {
+  activeTargetLabels: GristLabelType[]
+  activeTopcisLabels: GristLabelType[]
+  facility: MinimalRecordType
+  isFilteredByTarget: boolean
+  isFilteredByTopic: boolean
+}
+
+export const isFaclilityActive = ({
+  activeTargetLabels,
+  activeTopcisLabels,
+  isFilteredByTarget,
+  isFilteredByTopic,
+  facility,
+}: IsFaclilityActivePropType): boolean => {
+  if (!isFilteredByTopic && !isFilteredByTarget) return true
+  const hasSomeOfTheActiveTopics =
+    activeTopcisLabels?.some((tag) => facility.labels.includes(tag.id)) || false
+  const hasAnActiveTarget =
+    (activeTargetLabels[0] &&
+      facility.labels.includes(activeTargetLabels[0].id)) ||
+    false
+
+  if (!isFilteredByTopic && isFilteredByTarget) return hasAnActiveTarget
+  if (isFilteredByTopic && !isFilteredByTarget) return hasSomeOfTheActiveTopics
+  return hasSomeOfTheActiveTopics && hasAnActiveTarget
+}
+
+export const getActiveLabelGroups = (labels: GristLabelType[]) => ({
+  activeTopcisLabels: labels.filter(
+    (f) => f.isActive && f.fields.group2 !== 'zielpublikum'
+  ),
+  activeTargetLabels: labels.filter(
+    (f) => f.isActive && f.fields.group2 === 'zielpublikum'
+  ),
+})
+
+export const getFilteredFacilities = ({
```

Comment:
Since this is a pure function and we don't oinly use it in the map context, we could move it to a more general util, such as `src/lib/getFilteredFacilities.ts` or something like that. Same for the `isFaclilityActive` check.

Reply:
I did extract it to a util called facilityFilterUtil.ts! 👍